File: Binary Search/Hard/LET_no154_Find Minimum in Rotated Sorted Array II.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0, right = nums.size() - 1;
        while (left + 1 < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] < nums[right])
                right = mid;
            else if (nums[mid] == nums[right])
                right--;
            else
                left = mid;
        }

        return nums[left] < nums[right] ? nums[left] : nums[right];
    }
};

File: Binary Search/Hard/LET_no154_Find Minimum in Rotated Sorted Array II.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1
        while left + 1 < right:
            mid = (left + right) // 2
            if nums[mid] == nums[right]:
                right -= 1
            elif nums[mid] < nums[right]:
                right = mid
            else:
                left = mid
        
        return nums[left] if nums[left] < nums[right] else nums[right]